src/assets/CollectiblesController.ts

```diff
@@ -675,6 +671,26 @@ export class CollectiblesController extends BaseController<
         external_link && { externalLink: external_link },
       );
 
+      const existingEntry = collectibleContracts.find(
+        (collectibleContract) =>
+          collectibleContract.address.toLowerCase() === address.toLowerCase(),
+      );
+
+      if (existingEntry) {
+        if (compareCollectibleContract(newEntry, existingEntry)) {
+          const indexToRemove = collectibleContracts.findIndex(
+            (collectibleContract) =>
+              collectibleContract.address.toLowerCase() ===
+              address.toLowerCase(),
+          );
+          if (indexToRemove !== -1) {
+            collectibleContracts.splice(indexToRemove, 1);
```

Comment:
This potentially changes `allCollectibleContracts` in state. What do you think about making a copy of `collectibleContracts` and removing this fron the copy?

Reply:
done here: https://github.com/MetaMask/controllers/pull/680/commits/2bef09890f6df2392729b8cf4b99623af9322c0f